src/App/Fossa/Analyze/VulnerabilityReport.hs

```diff
@@ -0,0 +1,138 @@
+import Data.Maybe qualified
+import Data.Set qualified as Set
+import Data.String.Conversion (toString, toText)
+import Data.Text (Text)
+import Data.Text qualified as Text
+import DepTypes (Dependency (..), VerConstraint)
+import Effect.Logger (
+  Logger,
+  logInfo,
+ )
+import Graphing (directList, getRootsOf, vertexList)
+import Path (File, SomeBase)
+import Prettyprinter (Pretty (pretty), vsep)
+import Srclib.Converter (verConstraintToRevision)
+import Text.PrettyPrint.Boxes qualified as Boxes
+
+data VulnerableDependency = VulnerableDependency
+  { vdName :: Text
+  , isVulnerable :: Maybe VerConstraint -> Bool
+  }
+
+data VulnerableDependencyOrigin
+  = VulnerableDependencyManifestFiles [SomeBase File]
+  | VulnerableDependencyRootDependency [Dependency]
+  deriving (Eq, Ord)
+
+instance Show VulnerableDependencyOrigin where
+  show (VulnerableDependencyManifestFiles files) = toString $ Text.intercalate ", " (map (toText . show) files)
+  show (VulnerableDependencyRootDependency deps) = toString (Text.intercalate ", " $ map dependencyName deps)
+
+data VulnerableReportItem = VulnerableReportItem
+  { vriDep :: Dependency
+  , vriDepOriginDep :: VulnerableDependencyOrigin
+  }
+  deriving (Eq, Ord)
+
+-- HACK - Please FIX ASAP!
+-- Since we do not have built in version evaluator system, nor we parse version text into representative form.
+-- This is simplest of hack, ideally each versions are listed as opposed to providing evaluator functions.
+allVersionsBelow2_17 :: Maybe VerConstraint -> Bool
+allVersionsBelow2_17 version = case verConstraintToRevision =<< version of
+  Nothing -> False
+  Just ver -> any (`Text.isPrefixOf` ver) below2_17
+    where
+      below2_17 :: [Text] =
+        "1." : map (("2." <>) . toText . show) (takeWhile (<= (16 :: Int)) $ iterate (+ 1) 0)
+
+getVulnerableDeps :: [VulnerableDependency]
+getVulnerableDeps =
+  [ VulnerableDependency "org.apache.logging.log4j:log4j-core" allVersionsBelow2_17
+  , VulnerableDependency "org.apache.logging.log4j:log4j" allVersionsBelow2_17
+  , VulnerableDependency "log4j:log4j" (allVersionsBelow2_17)
+  ]
+
+logVulnerabilityReport :: (Has Diag.Diagnostics sig m, Has Logger sig m) => [ProjectResult] -> m ()
+logVulnerabilityReport = traverse_ printProjectReport
+
+printProjectReport :: (Has Diag.Diagnostics sig m, Has Logger sig m) => ProjectResult -> m ()
+printProjectReport ProjectResult{..} =
+  logInfo $
+    vsep
+      [ ""
+      , "Project Path: " <> (pretty . show $ projectResultPath)
+      , "Dependencies: "
+      , ""
+      , pretty renderedReport
+      ]
+  where
+    renderedReport :: String
+    renderedReport =
+      Boxes.render
+        . Boxes.hsep 2 Boxes.left
+        . map (Boxes.vcat Boxes.left . map Boxes.text)
+        $ table
+
+    table :: [[String]]
+    table =
+      List.transpose
+        ( [ ["Name", "Version", "Vulnerable?", "Type", "Environment", "Origin (manifest files if direct dependency, otherwise responsible direct dependency)"]
```

Comment:
> manifest files if direct dependency, otherwise responsible direct dependency

So I was thinking about transitive inclusions.  I think the ideal UX reports the manifest files of each item in the list of direct dependencies where a vulnerable dependency is reachable from that direct dep.

Example:
```raw
A --> B -> C -> log4j
           ^
          /
D --------
```

This would report log4j, citing `A` and `D` as the direct inclusions that brought it in, and the manifest files that included them (if possible to track).  However, I would timebox this to one hour and throw it away if it can't be done in that time. (I think we don't track deps with their exact origin files, so that may be impossible to do in a timely manner)